src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -0,0 +1,213 @@
+package me.zeroeightsix.kami.feature.module
+
+import imgui.ImGui
+import imgui.StyleVar
+import imgui.dsl
+import imgui.internal.sections.ItemFlag
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+import me.zero.alpine.listener.EventHandler
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.conditionalWrap
+import me.zeroeightsix.kami.event.TickEvent
+import me.zeroeightsix.kami.mixin.extend.setCooldown
+import me.zeroeightsix.kami.setting.ImGuiExtra
+import me.zeroeightsix.kami.util.InstrumentMap
+import me.zeroeightsix.kami.util.MidiParser
+import me.zeroeightsix.kami.util.Note
+import me.zeroeightsix.kami.util.Viewblock.getIrreplaceableNeighbour
+import me.zeroeightsix.kami.util.plus
+import me.zeroeightsix.kami.util.text
+import net.minecraft.block.Blocks
+import net.minecraft.block.NoteBlock.INSTRUMENT
+import net.minecraft.block.NoteBlock.NOTE
+import net.minecraft.block.enums.Instrument
+import net.minecraft.client.network.ClientPlayerEntity
+import net.minecraft.client.world.ClientWorld
+import net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket
+import net.minecraft.util.Hand
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Vec3d
+import java.io.File
+import java.util.ArrayList
+import java.util.TreeMap
+
+@Module.Info(
+    name = "Notebot",
+    category = Module.Category.MISC,
+    description = "Plays music with Noteblocks; put songs as .mid files in .mincraft/songs"
+)
+object Notebot : Module() {
+
+    private var lastNote: Long = 0L
+    private var elapsed: Long = 0L
+    var playingSong = false
+
+    private var noteSequence: TreeMap<Long, ArrayList<Note>> = TreeMap()
+    var instrumentMap = InstrumentMap()
+
+    var channelMap: TreeMap<Int, Instrument> = TreeMap()
+
+
+    /* This is a bad way to have channels. In the future, I would like for there to be some way to display a map.*/
+    @Setting(name = "Channel 0")
+    var channelZero = Instrument.HAT
+
+    @Setting(name = "Channel 1")
+    var channelOne = Instrument.HAT
+
+    @Setting(name = "Channel 2")
+    var channelTwo = Instrument.HAT
+
+    @Setting(name = "Channel 3")
+    var channelThree = Instrument.HAT
+
+    @Setting(name = "Channel 4")
+    var channelFour = Instrument.HAT
+
+    @Setting
+    var hatAlwaysAsFSharp = true
+
+    @Setting(comment = "Name of MIDI file in the songs folder of your .minecraft directory")
+    @ImGuiExtra.Post("renderButtons")
+    var songName = "tetris_theme.mid"
+
+    fun renderButtons() {
+        dsl.button("Discover") {
+            discoverSong()
+        }
+        dsl.button("Load") {
+            loadSong()
+        }
+
+        ImGui.pushItemFlag(ItemFlag.Disabled.i, true)
+        ImGui.pushStyleVar(StyleVar.Alpha, ImGui.style.alpha * 0.5f)
+
+        ImGui.popItemFlag()
+        ImGui.popStyleVar()
```

Comment:
There's no reason to push/pop disabled flag and colour if you don't draw anything inbetween.

Reply:
If I don't do that, the "Play" button is always greyed out.

Reply:
My bad. I misunderstood and only removed the last two lines.